private/buf/cmd/buf/command/push/push.go

```diff
@@ -184,21 +247,32 @@ func run(
 		}),
 	)
 	if err != nil {
-		if connect.CodeOf(err) == connect.CodeAlreadyExists {
-			if _, err := container.Stderr().Write(
-				[]byte("The latest commit has the same content; not creating a new commit.\n"),
-			); err != nil {
-				return err
-			}
-			return nil
-		}
-		return err
+		return nil, err
 	}
-	if resp.Msg.LocalModulePin == nil {
-		return errors.New("Missing local module pin in the registry's response.")
+	return resp.Msg.LocalModulePin, nil
+}
+
+func manifestAndFilesBlobs(ctx context.Context, sourceBucket storage.ReadBucket) (*modulev1alpha1.Blob, []*modulev1alpha1.Blob, error) {
```

Comment:
Nit: This func would be useful for the backend to remove some duplicated code, could you move it to the `manifest` pkg and export it? Not blocker though.

Reply:
Moved to `manifest.BucketToManifestAndFileBlobs`.